atmo/appsource/appsource.go

```diff
@@ -13,37 +13,48 @@ var (
 	ErrAuthenticationFailed = errors.New("failed to authenticate")
 )
 
-// Meta describes the metadata for an App
+// Meta describes the metadata for an App.
 type Meta struct {
 	Identifier string `json:"identifier"`
 	AppVersion string `json:"appVersion"`
+	Domain     string `json:"domain"`
 }
 
 type AppSource interface {
-	// Start indicates to the AppSource that it should prepare for app startup
+	// Start indicates to the AppSource that it should prepare for app startup.
 	Start(options.Options) error
-	// Runnables returns all of the available Runnables
-	Runnables() []directive.Runnable
+
+	// Runnables returns all of the available Runnables.
+	Runnables(string, string) []directive.Runnable
```

Comment:
```suggestion
	Runnables(identifier, version string) []directive.Runnable
```

Can we add the argument names here?

Reply:
we could, but should we? I'm torn, because
- on one hand, definitions of interfaces shouldn't describe what the data is, just what it looks like (so it should only care that it's a `string`, and should not care that that string is an identifier. The implementation should care that that is an identifier)
- - on the other hand it would be nice to know how an interface is used, and what kind of data is being passed to the implementations, in which case having the argument names in the interface definition helps with precision to know what the purpose is
  - though another way to solve that is to add those to the method comments

What do you think?

Reply:
But also https://www.integralist.co.uk/posts/go-interfaces/#name-your-interface-arguments, which is _for_ naming arguments.

Reply:
Fixed in https://github.com/suborbital/atmo/pull/121/commits/3efcc031a5070b24afb655d39097cc1fcc04d979